docs/design/nab_and_nar_status_update.md

```diff
@@ -0,0 +1,192 @@
+# Controller Workflow: NonAdminBackup and NonAdminRestore Status Update
+
+## Overview
+
+This document outlines the design around updating NonAdminBackup (NAB) and NonAdminRestore (NAR) objects' Statuses.
+
+## NonAdminBackup and NonAdminRestore Status
+
+The `status` field of NAB and NAR objects contains the following fields:
+- `phase`
+- `conditions`
+- `veleroBackup` for NAB and `veleroRestore` for NAR, which contains name, namespace and status of the related Velero object.
```

Comment:
Just to be on a very clear path. Should we have as well:
"which contains name, namespace, status and generated non-admin UUID of the corresponding Velero object."

Reply:
We need to add it somewhere. Either as label or in Status. Adding in label as we discussed will require webhook prevent users to modify the label to be set/changed from the user. Adding this as status will make our life easier by not requiring webhook to prevent that, because it's the controller only who will be setting that UUID as part of Status part of the object.